rustls/src/server/hs.rs

```diff
@@ -414,9 +384,16 @@ impl State for ExpectClientHello {
             None => None,
         };
 
-        if !self.done_retry {
-            // save only the first SNI
-            save_sni(conn, sni.clone());
+        // save only the first SNI
+        if let (Some(sni), false) = (&sni, self.done_retry) {
+            // Save the SNI into the session.
+            // The SNI hostname is immutable once set.
+            assert!(conn.sni.is_none());
+            conn.sni = Some(sni.clone());
+        } else if conn.sni != sni {
```

Comment:
This is a case-insensitive comparison, right? Do you mean to check that the SNI didn't change at all?

Reply:
I guess so.